core/src/mindustry/logic/LAssembler.java

```diff
@@ -99,6 +99,8 @@ public static String write(Seq<LStatement> statements){
         String[] lines = data.split("\n");
         int index = 0;
         for(String line : lines){
+            //ignore empty lines
+            if(line.isEmpty()) continue;
```

Comment:
it's cleaner to merge this with the next continue line, so I'd rather do this myself.